server/service/src/sync/translations/name_oms_fields.rs

```diff
@@ -0,0 +1,115 @@
+use repository::{
+    ChangelogRow, ChangelogTableName, NameOmsFieldsRow, NameRowRepository, StorageConnection,
+    SyncBufferRow,
+};
+
+use crate::sync::translations::name::NameTranslation;
+
+use super::{
+    PullTranslateResult, PushTranslateResult, SyncTranslation, ToSyncRecordTranslationType,
+};
+
+// Needs to be added to all_translators()
+#[deny(dead_code)]
+pub(crate) fn boxed() -> Box<dyn SyncTranslation> {
+    Box::new(NameOmsFieldsTranslation)
+}
+
+pub(super) struct NameOmsFieldsTranslation;
+impl SyncTranslation for NameOmsFieldsTranslation {
+    fn table_name(&self) -> &str {
+        "name_oms_fields"
+    }
+
+    fn pull_dependencies(&self) -> Vec<&str> {
+        vec![NameTranslation.table_name()]
+    }
+
+    fn change_log_type(&self) -> Option<ChangelogTableName> {
+        Some(ChangelogTableName::NameOmsFields)
+    }
+
+    fn try_translate_from_upsert_sync_record(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<PullTranslateResult, anyhow::Error> {
+        let upsert_record = PullTranslateResult::upsert(serde_json::from_str::<NameOmsFieldsRow>(
+            &sync_record.data,
+        )?);
+        Ok(upsert_record)
+    }
+
+    fn should_translate_to_sync_record(
+        &self,
+        row: &ChangelogRow,
+        r#type: &ToSyncRecordTranslationType,
+    ) -> bool {
+        match r#type {
+            ToSyncRecordTranslationType::PullFromOmSupplyCentral => {
+                self.change_log_type().as_ref() == Some(&row.table_name)
+            }
+            ToSyncRecordTranslationType::PushToOmSupplyCentral => {
+                self.change_log_type().as_ref() == Some(&row.table_name)
+            }
+            _ => false,
+        }
+    }
+
+    fn try_translate_to_upsert_sync_record(
+        &self,
+        connection: &StorageConnection,
+        changelog: &ChangelogRow,
+    ) -> Result<PushTranslateResult, anyhow::Error> {
+        let name_row = NameRowRepository::new(connection)
+            .find_one_by_id(&changelog.record_id)?
+            .ok_or(anyhow::Error::msg(format!(
+                "Name row ({}) not found",
+                changelog.record_id
+            )))?;
+
+        if name_row.deleted_datetime.is_some() {
+            return Ok(PushTranslateResult::Ignored(
+                "Ignore pushing soft deleted name".to_string(),
+            ));
+        }
+
+        let row = NameRowRepository::new(connection)
+            .find_one_oms_fields_by_id(&changelog.record_id)?
+            .ok_or(anyhow::Error::msg(format!(
+                "NameOmsFields row ({}) not found",
```

Comment:
```suggestion
                "Name row ({}) not found for Name OMS Fields translation",
```
Might be a little bit more helpful if it every happens?

Reply:
Ah nice thanks [e94c800](https://github.com/msupply-foundation/open-msupply/pull/4103/commits/e94c8004cb38f13a1ca97703b919bd309ce56240)